File: next_prime.c
#include <stdio.h>
#include <math.h>

int isPrime(int n) {
	if (n == 2) {
		return 1;
	} else if (n % 2 == 0) {
		return 0;
	}
	for (int i = 3; i <= pow(n, 0.5); i += 2) {
		if (n % i == 0)
			return 0;
	}
	
	return 1;
}

int generatePrime(int currentPrime) {
	
	while (1) {
		currentPrime++;
		if (isPrime(currentPrime)) {
			break;
		}
		
	}
	return currentPrime;
}

void main(void) {
	int currentPrime = 2;
	char answer;
	
	while (1) {
		printf("Would you like to the see the next prime number? [y/n]\t");
		scanf(" %c", &answer);
		if (answer == 'Y' || answer == 'y') {
			printf("%d\n", currentPrime);
			currentPrime = generatePrime(currentPrime);
		} else {
			printf("\n");
			break;
		}
	}
}

File: next_prime.py
def isPrime(x):
    """
    Checks whether the given
    number x is prime or not
    """

    if x == 2: return True

    if not x % 2: return False

    for i in range(3, int(x**0.5)+1, 2):
        if x % i == 0:
            return False

    return True


def generatePrime(currentPrime):
    """
    Returns the next prime
    after the currentPrime
    """

    newPrime = currentPrime + 1

    while True:

        if not isPrime(newPrime):
            newPrime += 1
        else:
            break

    return newPrime


def main():

    currentPrime = 2

    while True:

        answer = input('Would you like to see the next prime? (Y/N) ')

        if answer.lower().startswith('y'):
            print(currentPrime)
            currentPrime = generatePrime(currentPrime)

        else:
            break

if __name__ == '__main__':
	main()
